server/db/helper.js

```diff
@@ -114,6 +114,25 @@ const getUserBadges = (userId) => UserBadge.findAll({
 });
 
 
+const getCompletedCourse = (userId) => UserBadge.findAll({
+  where: {
+    idUser: userId
+  },
+
+  attributes: ['idBadge']
+})
+.then((courses) => {
+  let courseId = courses.map((course) => course.dataValues.idBadge)
+  
+  return Course.findAll({
+    where: {
+      id: courseId
```

Comment:
Since the id that you are retrieving is the badge id, it will not always match the course id so you need to compare it to the course's idBadge field